api/internal/managers/linux/installation/config.go

```diff
@@ -233,3 +236,62 @@ func (m *installationManager) ConfigureHost(ctx context.Context, rc runtimeconfi
 
 	return nil
 }
+
+// CalculateRegistrySettings calculates registry settings for airgap installations
+func (m *installationManager) CalculateRegistrySettings(ctx context.Context, releaseData *release.ReleaseData) (*types.RegistrySettings, error) {
+	// Only return registry settings for airgap installations
+	if m.airgapBundle == "" {
+		return nil, nil
+	}
+
+	// Get the current config to access service CIDR
+	config, err := m.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get installation config: %w", err)
+	}
+
+	serviceCIDR := config.ServiceCIDR
+	if serviceCIDR == "" {
+		// Fallback to runtime config if not set in installation config
+		rc, err := runtimeconfig.NewFromDisk()
```

Comment:
runtimeconfig should be passed as an argument to the function similar to `SetConfigDefaults` and `ConfigureHost`